storage/sql/common/flag.go

```diff
@@ -71,32 +72,43 @@ func (s *Store) GetFlag(ctx context.Context, key string) (*flipt.Flag, error) {
 }
 
 // ListFlags lists all flags
-func (s *Store) ListFlags(ctx context.Context, opts ...storage.QueryOption) ([]*flipt.Flag, error) {
-	var (
-		flags []*flipt.Flag
-
-		query = s.builder.Select("\"key\", name, description, enabled, created_at, updated_at").
-			From("flags").
-			OrderBy("created_at ASC")
-	)
-
+func (s *Store) ListFlags(ctx context.Context, opts ...storage.QueryOption) (storage.ResultSet[*flipt.Flag], error) {
 	params := &storage.QueryParams{}
 
 	for _, opt := range opts {
 		opt(params)
 	}
 
+	var (
+		flags   []*flipt.Flag
+		results = storage.ResultSet[*flipt.Flag]{}
+
+		query = s.builder.Select("\"key\", name, description, enabled, created_at, updated_at").
+			From("flags").
+			OrderBy(fmt.Sprintf("created_at %s", params.Order))
+	)
+
 	if params.Limit > 0 {
-		query = query.Limit(params.Limit)
+		query = query.Limit(params.Limit + 1)
 	}
 
-	if params.Offset > 0 {
-		query = query.Offset(params.Offset)
+	var offset uint64
+
+	if params.PageToken != "" {
+		var token PageToken
+		if err := json.NewDecoder(bytes.NewBufferString(params.PageToken)).Decode(&token); err != nil {
```

Comment:
Take it or leave it:
```suggestion
		if err := json.Unmarshal([]byte(params.PageToken), &token); err != nil {
```